### DISCUSSION TITLE
Bun without Typescript

### DISCUSSION THREAD

**Discussion by sprkv5** [oven-sh/bun#5168] [General]
I want to use plain JavaScript (via ReScript). Thus I don't want Typescript to be added to the `peerDependencies` for a new bun project created via `bun init`. Can there be a mode to support languages that compile to JavaScript? This also means, having an equivalent of bun-types for ReScript, which can be built by that community. So bun-types might also not be required. I want to initiate a discussion on how ReScript/ReasonML/etc. could be minimally supported as secondary languages.

```sh
$ bun init
bun init helps you get started with a minimal project and tries to guess sensible defaults. Press ^C anytime to quit

package name (test): 
entry point (index.ts): index.js

Done! A package.json file was saved in the current directory.
 + index.js
 + jsconfig.json (for editor auto-complete)
 + README.md

To get started, run:
  bun run index.js

$ vi package.json
 {
  "name": "test",
  "module": "index.js",
  "type": "module",
  "devDependencies": {
    "bun-types": "latest"
  },
  "peerDependencies": {
    "typescript": "^5.0.0"
  }
}
```

**Comment by melroy89** [comment]
Yea, this should be an issue (feature request) instead of a discussion.

**Reply by sprkv5** [reply]
Thanks for the guidance!
